system-test/logging-winston.ts

```diff
@@ -157,3 +267,41 @@ describe('LoggingWinston', () => {
     });
   });
 });
+
+
+
+// polls for the entire array of entries to be greater than logTime.
+function pollLogs(logName: string, logTime: number, size = 1, timeout = 90000):
+    Promise<types.StackdriverEntry[]> {
+  const p = new Promise((resolve, reject) => {
+    const end = Date.now() + timeout;
+    loop();
+
+    function loop() {
+      setTimeout(() => {
+        logging.log(logName).getEntries(
+            {
+              pageSize: size,
+            },
+            (err: Error, entries: types.StackdriverEntry[]) => {
+              const {receiveTimestamp} =
+                  (entries[entries.length - 1].metadata || {}) as
+                  {receiveTimestamp: {seconds: number, nanos: number}};
+              const time = (receiveTimestamp.seconds * 1000) +
+                  (receiveTimestamp.nanos * 1e-6);
+
+              if (time >= logTime) {
+                return resolve(entries);
+              }
+
+              if (Date.now() > end) {
+                return reject(new Error('timeout'));
+              }
+              loop();
+            });
+      }, 500);
+    }
+  });
+
+  return p as Promise<types.StackdriverEntry[]>;
```

Comment:
Is it possible to declare `const p: Promise<types.StackdriverEntry[]>` initially and avoid this cast?

Reply:
i declared it that way but still gave me a squiggly red line until i also added the cast. maybe you can point me in the right direction?